plugins/redis/redis.cpp

```diff
@@ -195,29 +221,99 @@ protected :
     redisContext * context = nullptr;
     StringAttr options;
     StringAttr ip; //The default is set in parseOptions as "localhost"
-    unsigned serverIpPortPasswordHash;
+    unsigned serverIpPortPasswordHash = 0;
     int port = 6379; //Default redis-server port
     TimeoutHandler timeout;
     int database = 0; //NOTE: redis stores the maximum number of dbs as an 'int'.
 };
+class SubConnection : public Connection
+{
+    friend class SubConnectionContainer;
+public :
+    SubConnection(ICodeContext * ctx, const char * _options, const char * _ip, int _port, bool parseOptions, int _database, const char * password, unsigned _timeout) :
+        Connection(ctx, _options, _ip, _port, parseOptions, _database, password, _timeout) { };
+    virtual void subAndWaitForSinglePub(ICodeContext * ctx, const char * keyOrChannel, size_t & messageSize, char * & message, int _database, bool lockedKey);
 
-#if HIREDIS_VERSION_OK_FOR_CACHING
-static void releaseContext()
+protected :
+    virtual void selectDB(ICodeContext * ctx, int _database) { /*do nothing as pub/sub is database agnostic*/ };
+    virtual void subscribe(ICodeContext * ctx, const char * channel);
+    virtual void unsubscribe(ICodeContext * ctx, const char * password);
+    virtual bool isCorrectChannel(const redisReply * reply, const char * op) const;
+    virtual void freeContext();
+
+    int redisSetUnsubscribeTimeout();
+    int getUnsubscribeTimeout();
+    unsigned getUnsubscribeReadAttempts();
+
+    //--------------------DELETE----------------------------------------------------
+    template <class type> void set(ICodeContext * ctx, const char * key, type value, unsigned expire) = delete;
```

Comment:
this smells wrong.  If you need to delete lots of functions like this I suspect that it should not be derived from Connection.

What does SubConnection represent?  From the name I would expect a child, but it looks like it might be a subscription.  If so, the abbreviation is confusing.